Realm/RLMManagedDictionary.mm

```diff
@@ -0,0 +1,688 @@
+    }
+}
+
+template<typename Function>
+static auto translateErrors(__unsafe_unretained RLMManagedDictionary *const dictionary,
+                            Function&& f, NSString *aggregateMethod=nil) {
+    try {
+        return f();
+    }
+    catch (...) {
+        throwError(dictionary, aggregateMethod);
+    }
+}
+
+template<typename Function>
+static auto translateErrors(Function&& f) {
+    try {
+        return f();
+    }
+    catch (...) {
+        throwError(nil, nil);
+    }
+}
+
+static void changeDictionary(__unsafe_unretained RLMManagedDictionary *const dict,
+                             dispatch_block_t f) {
+    translateErrors([&] { dict->_backingCollection.verify_in_transaction(); });
+
+    RLMObservationTracker tracker(dict->_realm);
+    tracker.trackDeletions();
+    auto obsInfo = RLMGetObservationInfo(dict->_observationInfo.get(),
+                                         dict->_backingCollection.get_parent_object_key(),
+                                         *dict->_ownerInfo);
+    if (obsInfo) {
+        tracker.willChange(obsInfo, dict->_key);
+    }
+
+    translateErrors(f);
+}
+
+//
+// public method implementations
+//
+- (RLMRealm *)realm {
+    return _realm;
+}
+
+- (NSUInteger)count {
+    return translateErrors([&] {
+        return _backingCollection.size();
+    });
+}
+
+- (NSArray *)allKeys {
+    return translateErrors([&] {
+        NSMutableArray<id> *keys = [NSMutableArray array];
+        auto keyResult = _backingCollection.get_keys();
+        for (size_t i=0; i<keyResult.size(); i++) {
+            [keys addObject:RLMStringDataToNSString(keyResult.get<realm::StringData>(i))];
+        }
+        return keys;
+    });
+}
+
+- (NSArray *)allValues {
+    return translateErrors([&] {
+        NSMutableArray *values = [NSMutableArray array];
+        auto valueResult = _backingCollection.get_values();
+        RLMAccessorContext c(*_objectInfo);
+        for (size_t i=0; i<valueResult.size(); i++) {
+            [values addObject:valueResult.get(c, i)];
+        }
+        return values;
+    });
+}
+
+- (BOOL)isInvalidated {
+    return translateErrors([&] { return !_backingCollection.is_valid(); });
+}
+
+- (RLMClassInfo *)objectInfo {
+    return _objectInfo;
+}
+
+- (bool)isBackedByDictionary:(realm::object_store::Dictionary const&)dictionary {
+    return _backingCollection == dictionary;
+}
+
+- (BOOL)isEqual:(id)object {
+    return [object respondsToSelector:@selector(isBackedByDictionary:)] &&
+           [object isBackedByDictionary:_backingCollection];
+}
+
+- (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state
+                                  objects:(__unused __unsafe_unretained id [])buffer
+                                    count:(NSUInteger)len {
+    return RLMFastEnumerate(state, len, self);
+}
+
+#pragma mark - Object Retrieval
+
+- (nullable id)objectForKey:(id)key {
+    return translateErrors([&] {
+        [self.realm verifyThread];
```

Comment:
`try_get_any()` should be checking the thread

Reply:
It doesn't seem to throw where it should without it